### PR TITLE
What is Word2Vec

### LINKED ISSUE #1609: What is Word2Vec?
Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Proposed title of article
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article higher ranking.*

### Introduction paragraph(s):

How can we make computers understand us better using natural language?

Computers are good at understanding numbers. They are good at taking large matrices and applying linear algebra to those matrices. By converting words into vectors, we can make computers understand the meaning and context of words in natural language. 

It turns out that the word2vec algorithm can help us achieve this.

### Key takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article? Use this area to get your ideas down on the bulk of your article or tutorial.

By the end of this article, the reader should be able to understand:

1. How computers are able to understand words
2. What the Word2Vec algorithm entails
3. The workings of the Continuous Bag of Words (CBOW) and Continuous Skip-gram models used in the Word2Vec algorithm

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

1. [Efficient Estimation of Word Representations in Vector Space](https://arxiv.org/pdf/1301.3781.pdf)
2. [word2vec Parameter Learning Explained](https://arxiv.org/pdf/1411.2738.pdf)
3. [Tomas Mikolov](https://en.wikipedia.org/wiki/Tomas_Mikolov)

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by wilkistermumbi**
This Fixes #1609

### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you placed your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

File: articles/what-is-word2vec/index.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+2. The Word2Vec algorithm causes words that have similar contexts to have similar vector embeddings (be closer together) and words that have different contexts to be far apart.
+3. Skipgram is relatively slower than CBOW and usually works well with a large corpus of training data.
+4. CBOW is faster to train than skip-gram. 
+
+### Wrapping Up
+
+I hope this article has been helpful to you. Word2vec was an extremely influential algorithm in NLP until recently when it went out of fashion as far as research is concerned primarily with the rise of language models such as Google's [BERT](https://en.wikipedia.org/wiki/BERT_(language_model)) and [ELMo](https://allennlp.org/elmo). However, it is still very relevant.
+
+It is also important to note that word2vec isn't the only word embedding models that work well. Stanford's [GloVe](https://en.wikipedia.org/wiki/GloVe_(machine_learning)) and Facebook's [FastText](https://fasttext.cc/) are examples of other popular word embedding models you can explore that also works so great. Feel free to also read on it.
```
Comment (collins-a) [inline articles/what-is-word2vec/index.md:88]:
```suggestion
It is also important to note that word2vec isn't the only word embedding model that works well. Stanford's [GloVe](https://en.wikipedia.org/wiki/GloVe_(machine_learning)) and Facebook's [FastText](https://fasttext.cc/) are examples of other popular word embedding models you can explore that also work well. Feel free to also have a read on them.
```

**Review by collins-a — COMMENTED** [review]
@wilkistermumbi great first article. Good introduction to Word2vec. I liked the easy-to-follow distinction between CBOW and continuous skip-gram. A few adjustments and we're good to go.

**Review by collins-a — COMMENTED** [review]
Changes seem in order. This is ready for the final review

**Comment by hectorkambow** [comment]
@wilkistermumbi Great work and welcome to the EngEd program again. 

Please be sure to check out our [resources pages](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) if you haven't already for more neat resources to explore. 

cc @collins-a

**Pull request merged**